### PR TITLE
Avoid falling into an endless loop of default value updates

### LINKED ISSUE #3473: Qfield freeze / crash when adding relation
Hello, 

The app Qfield freeze or crash when I try to add information in a relation layer. 

I add 2 video to visualize it: 
Freeze:
https://user-images.githubusercontent.com/72577399/196347953-270a2ca9-7050-4291-a090-4f45bc60cf55.mp4

Crash:
https://user-images.githubusercontent.com/72577399/196347932-e40ea2e3-1d36-4e3a-a3ac-54d6f77a8bc0.mp4

And also a test project: 

[test_simpl.zip](https://github.com/opengisch/QField/files/9806899/test_simpl.zip)

Thank you.

### Desktop (please complete the following information)
 - OS: [Windows 10]
 - QGIS Version [ 3.26.0]
 - QFieldSync Version [e.g.4.2.0]

### Mobile (please complete the following information)
 - Device: [Samsung galaxy S9]
 - OS: [e.g. Android 11]
 - QField version: 
 [master-dev 16c92a]
[Qfield 2.4.1 (5d2ac1)]

### Additional information
* Problem started happening recently, didn't happen in an older version of QField: [Yes]
* Problem can be reliably reproduced, doesn't happen randomly: [Yes]

### PR DESCRIPTION
**Pull request by nirvn**
Should address part of #3473 (the crasher).

### CODE REVIEW

File: src/core/attributeformmodelbase.h [hunk 1]
```diff
@@ -81,7 +81,7 @@ class AttributeFormModelBase : public QStandardItemModel
 
     void flatten( QgsAttributeEditorContainer *container, QStandardItem *parent, const QString &parentVisibilityExpressions, QVector<QStandardItem *> &items, int currentTabIndex = 0, const QColor &color = QColor() );
 
-    void updateDefaultValues( int fieldIndex = -1 );
+    void updateDefaultValues( int fieldIndex = -1, bool fromUpdatedDefaultValue = false );
```
Comment (m-kuhn) [inline src/core/attributeformmodelbase.h:84]: `QVector<int> updatedFields= QVector<int>()`

File: src/core/attributeformmodelbase.cpp [hunk 2]
```diff
@@ -511,7 +511,10 @@ void AttributeFormModelBase::updateDefaultValues( int fieldIndex )
     if ( success && updatedValue != previousValue )
     {
       item->setData( defaultValue, AttributeFormModel::AttributeValue );
-      updateDefaultValues( fidx );
+      if ( !fromUpdatedDefaultValue )
```
Comment (m-kuhn) [inline src/core/attributeformmodelbase.cpp:514]:
```
if( !updatedFields.contains( fidx ) )
{
  updatedFilds.append( fidx )
```

Let's just support cascading default values. Even if it's probably not the best idea, why not ...

Comment (nirvn) [reply src/core/attributeformmodelbase.cpp:514]: @m-kuhn , that's exactly what I was thinking last night when not feeling satisfied about the bool solution

Comment (nirvn) [reply src/core/attributeformmodelbase.cpp]: @m-kuhn , done.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**